Realm/Tests/Swift/SwiftObjectInterfaceTests.swift

```diff
@@ -313,6 +313,16 @@ class SwiftObjectInterfaceTests: RLMTestCase {
         try! realm.commitWriteTransaction()
     }
 
+    func testObjectForPrimaryKey() {
+        let realm = RLMRealm.default()
+        realm.beginWriteTransaction()
```

Comment:
Should that transaction be committed or rolled back (or does it happen in a non-obvious way)?

Reply:
Good point - when the realm goes out of scope and is destroyed that should get rid of the transaction, and closing the transaction in one way or another isn't really the point of this test, but it might not hurt to explicitly close it like in the test above.

Reply:
Yeah, committing or cancelling the transaction is semantically unnecessary, but we log a warning if the Realm goes out of scope while in a transaction so we end the transaction in tests just to avoid the log spam.